django/contrib/admin/models.py

```diff
@@ -44,6 +44,27 @@ def log_action(
             change_message=change_message,
         )
 
+    def log_actions(self, user_id, queryset, action_flag, change_message=""):
+        """
+        Method to bulk_create multiple LogEntry objects
+        from a given queryset.
+        """
+        if isinstance(change_message, list):
+            change_message = json.dumps(change_message)
+        log_entry_list = [
+            self.model(
+                user_id=user_id,
+                content_type_id=ContentType.objects.get_for_model(obj).id,
+                object_id=str(obj.pk),
+                object_repr=repr(obj)[:200],
+                action_flag=action_flag,
+                change_message=change_message,
+            )
+            for obj in queryset
+        ]
+
+        return self.model.objects.bulk_create(log_entry_list)
```

Comment:
This will not work on all supported databases, because `bulk_create()` doesn't return IDs on SQLite < 3.35, MariaDB < 10.5, Oracle, and probably on most of databases supported via 3rd party backends (as [documented](https://docs.djangoproject.com/en/4.2/ref/models/querysets/#bulk-create)).

Reply:
ah keen eye there felix

I wonder whether we should just leave the objs without pks; I can't think of a robust way to populate them in the absence of a pk.

We could leave it up to the user to determine the best way to fetch from the logentry table?  An admonition in docs as well?

Reply:
It should be enough to document that it uses `bulk_create()` and has the same caveats for returned `LogEntry()` objects.

Reply:
Awesome @Akash-Kumar-Sen ok looks like you'll need to update the tests to accommodate this caveat

Reply:
Done. Please let me know if any other changes are required.

Reply:
@Akash-Kumar-Sen PS also if you've done all your changes it's helpful to change the PR state from "Draft" to "Ready to review" 😊

Reply:
There is still a lot of things to cover, for example, docs changes, release notes, deprecation of current `log*` methods in a backward compatible way, etc.

Reply:
Changed it to draft again for working on a query count assertion to make sure we're solving the originally reported problem as proposed by you. But I think the number of queries will be different for different database backends, let me know if that is necessary.